src/auth/ejabberd_auth.erl

```diff
@@ -61,15 +61,18 @@
 
 %% Hook handlers
 -export([remove_domain/3]).
+-export([does_user_exist/3]).
 
 -include("mongoose.hrl").
 -include("jlib.hrl").
 
 -export_type([authmodule/0,
-              passterm/0]).
+              passterm/0,
+              does_user_exist/0]).
 
 -type authmodule() :: module().
 -type passterm() :: binary() | mongoose_scram:scram_tuple() | mongoose_scram:scram_map().
+-type does_user_exist() :: #{result := boolean(), cached := boolean(), '_' := '_'}.
```

Comment:
do we really use this `cached` field anywhere?

Reply:
On the third subscriber, to know that we don't need to cache it again.

Reply:
don't get it, do you want to have a multi layer cache? I don't think that we need it.

Reply:
If the first subscriber returned true, the third (which is the same module, mod_cache_users) should not cache it again, so the third subscriber (put_user_into_cache or sth like that), needs to know who said it's true, the cache, or the backend, to know if it should cache it or not.

Reply:
again, why don't we just stop the execution of the hook?